lib/libc/minimal/source/math/sqrt.c

```diff
@@ -5,19 +5,50 @@
  */
 
 #include <math.h>
+#include <stdint.h>
+#include <zephyr/sys/util.h>
 
-double sqrt(double value)
+#define MAX_D_ITTERATIONS	8  /* usually converges in 5 loops */
+				   /* this ensures we break out of the loop */
+#define MAX_D_ERROR_COUNT	5  /* when result almost converges, stop */
+#define EXP_MASK64	GENMASK64(62, 52)
+
+double sqrt(double square)
 {
-	double sqrt = value / 3;
 	int i;
+	int64_t	exponent;
+	double root;
+	double last;
+	int64_t *p_square = (int64_t *)&square;
```

Comment:
Should these be uint64_t instead? I was just looking at the termination condition and wondering what would happen if the two values differed in sign (which I don't think is possible), if they did, then you'd exit the loop even if they were very different in magnitude.

Reply:
root needs to be double for the main algorithm to work. Making last a double means that the compiler doesn't do any conversions (just a move) when assigning last = root.